fermipy/roi_model.py

```diff
@@ -2171,3 +2176,29 @@ def write_fits(self, fitsfile):
         hdu_data = fits.table_to_hdu(tab)
         hdus = [fits.PrimaryHDU(), hdu_data]
         fits_utils.write_hdus(hdus, fitsfile)
+
+    def write_ds9region(self, region, free='box',fixed='cross',frame='fk5',color='green',header=True):
+        """Create a ds9 compatible region file from the ROI
+        :param region: name of the region file (string)
+        :param free: one of the supported ds9 point symbols, used for free sources, see here: http://ds9.si.edu/doc/ref/region.html
+        :param fixed: as free, but for fixed sources
+        :param frame: typically fk5, more to be implemented
+        :param color: color used for the symbols
+        :param header: if True, will prepend a global header line.
+        :return: None.
+        """
+        allowed_symbols = ['circle','box','diamond','cross','x','arrow','boxcircle']
+        # adding some checks.
+        assert free in allowed_symbols, "symbol %s not supported"%free
+        assert fixed in allowed_symbols, "symbol %s not supported"%fixed
+        with open(region,'w') as fo:
+            if header:
+                fo.write("global color=%s\n"%color)
+            for src in self.get_sources():
+                if not isinstance(src,Source): continue
+                ra, dec = src.radec
+                fo.write("%s; point( %1.5f, %1.5f) # point=%s text={%s} color=%s \n"%(frame,ra, dec,
+                                                                                      free if src.is_free else fixed,
+                                                                                      src.name,
+                                                                                      color) )
+            fo.close()
```

Comment:
you shouldn't close explicitly the file if you use a `with` statement, the file will be closed automatically when the `with` indentation is finished.

Reply:
changed in last commit.